controller/state.go

```diff
@@ -409,7 +409,7 @@ func (m *appStateManager) CompareAppState(app *v1alpha1.Application, project *ap
 
 	for _, liveObj := range liveObjByKey {
 		if liveObj != nil {
-			appInstanceName := kubeutil.GetAppInstanceLabel(liveObj, appLabelKey)
+			appInstanceName := kubeutil.GetAppInstanceIdentifier(liveObj, appLabelKey)
```

Comment:
Please make sure to check for annotation and label in

https://github.com/argoproj/argo-cd/blob/422a26e8d8c5f4ab2da3cd65883426901de53d37/controller/cache/cache.go#L259


Controller should cache whole resource manifest if either annotation or label present.